src/api/cvc4cpp.cpp

```diff
@@ -2724,10 +2862,26 @@ OpTerm Solver::mkOpTerm(Kind kind, Kind k) const
 OpTerm Solver::mkOpTerm(Kind kind, const std::string& arg) const
 {
   CVC4_API_SOLVER_TRY_CATCH_BEGIN;
-  CVC4_API_KIND_CHECK_EXPECTED(kind == RECORD_UPDATE_OP, kind)
-      << "RECORD_UPDATE_OP";
-
-  return *mkValHelper<CVC4::RecordUpdate>(CVC4::RecordUpdate(arg)).d_expr.get();
+  CVC4_API_KIND_CHECK_EXPECTED(
+      (kind == RECORD_UPDATE_OP) || (kind == DIVISIBLE_OP), kind)
+      << "RECORD_UPDATE_OP or DIVISIBLE_OP";
+  OpTerm res;
+  switch (kind)
+  {
+    case RECORD_UPDATE_OP:
+      res = *mkValHelper<CVC4::RecordUpdate>(CVC4::RecordUpdate(arg))
+                 .d_expr.get();
+      break;
+    case DIVISIBLE_OP:
+      res = *mkValHelper<CVC4::Divisible>(CVC4::Divisible(CVC4::Integer(arg)))
+                 .d_expr.get();
+      break;
+    default:
```

Comment:
This case can never occur, kind is guarded in the check above. You can safely convert this `switch` to an `if`. For API functions we prefer having argument checks at the top of the function, hence the preference to convert this to `if` and keep the guard above.

Reply:
No worries, good catch! Fixed with f0cd7ef0844e57cd483907946c0f909b9b49af1f. I turned it into an if/else without checking the second case because there's only two possibilities. I also took at the assertion that `res` is not null because that should never happen.

One last question: a possible issue is that the user passes in a string that doesn't represent a valid `Integer`. Should I catch that exception and do something special or just let the API try/catch macro handle it?

Reply:
In general we would want to always catch such cases on the API level (and then guard internally with assertions that no invalid argument is passed internally -- think of API guards as a 'fence').

With Integer it's a bit of a special case, unfortunately, since we support two different back end implementations (GMP, CLN), and they do not necessarily agree on what is (in)valid input. So, yes, we generally want to explicitly guard against it in the API, but since we use a 3rd-party backend we have no control over and which throws an exception anyways, we delegate it to the back end and let the macro handle it.

But good thing you brought this up, because there is one thing we actually need to do here due to the above mentioned differences. Add a guard like this one here:
https://github.com/CVC4/CVC4/blob/master/src/api/cvc4cpp.cpp#L2067

And let me also add what I wrote above as documentation of the file. Will submit PR.

Reply:
Cool, added the guard in b3068b68424b2bd0be4d94dd852be7e8da25a17a.